myapp/app/controllers/application_controller.rb

```diff
@@ -1,2 +1,38 @@
 class ApplicationController < ActionController::Base
+
+  around_action :switch_locale
+
+  def switch_locale(&action)
+    locale = params[:locale] || I18n.default_locale
+    I18n.with_locale(locale, &action)
+  end
+
+  # keep locale in urls
+  def default_url_options
+    { locale: I18n.locale }
+  end
+
+
+  # unless Rails.env.development?
+    rescue_from Exception,                        with: :_render_500
+    rescue_from ActiveRecord::RecordNotFound,     with: :_render_404
+    rescue_from ActionController::RoutingError,   with: :_render_404
+  # end
+
+  def routing_error
+    raise ActionController::RoutingError, params[:path]
+  end
+
+  private
+    def _render_404(e = nil)
+      Rails.logger.info "Rendering 404 with exception: #{e.message}" if e
+
+      render 'errors/404', status: :not_found
+    end
+
+    def _render_500(e = nil)
+      Rails.logger.error "Rendering 500 with exception: #{e.message}" if e
+
+      render 'errors/500', status: :internal_server_error
+    end
```

Comment:
```suggestion
  private
  
  def _render_404(e = nil)
    Rails.logger.info "Rendering 404 with exception: #{e.message}" if e

    render 'errors/404', status: :not_found
  end

  def _render_500(e = nil)
    Rails.logger.error "Rendering 500 with exception: #{e.message}" if e

    render 'errors/500', status: :internal_server_error
  end
```
In Rakuma, indent level of `private` and methods below it should be the same

Reply:
Seems like there is no description about [this](https://github.com/Fablic/training/blob/m-alamin/steps_jp.md#:~:text=Gemfile%20%E3%81%A7%20RuboCop%EF%BC%88Ruby%E3%81%AE%E9%9D%99%E7%9A%84%E3%82%B3%E3%83%BC%E3%83%89%E8%A7%A3%E6%9E%90%E3%83%84%E3%83%BC%E3%83%AB%EF%BC%89%20%E3%81%8B%20fablicop%E3%82%92%E3%82%A4%E3%83%B3%E3%82%B9%E3%83%88%E3%83%BC%E3%83%AB%E3%81%97%E3%81%BE%E3%81%97%E3%82%87%E3%81%86%E3%80%82%E8%A8%AD%E5%AE%9A%E3%82%84%E3%82%B3%E3%83%9E%E3%83%B3%E3%83%89%E5%AE%9F%E8%A1%8C%E6%96%B9%E6%B3%95%E3%81%AF%E5%90%84README%E3%81%8B%E3%82%89%E7%A2%BA%E8%AA%8D%E3%81%97%E3%81%BE%E3%81%97%E3%82%87%E3%81%86%E3%80%82) in English ver README. You should install `fablicop` gem to follow linting rules in Rakuma 😄

Reply:
fixed